sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1291,4 +1291,19 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       Seq(1 -> "a").toDF("i", "j").filter($"i".cast(StringType) === "1"),
       Row(1, "a"))
   }
+
+  test("SPARK-12982: Add table name validation in temp table registration") {
+    val rows = List(Row("foo"), Row("bar"))
```

Comment:
We can be a bit more concise (also see the comment in the previous PR): `val df = Seq(("one", "two")).toDF("a", "b")`